src/Arcus.Scripting.Storage.FileShare/Scripts/Create-AzFileShareStorageFolder.ps1

```diff
@@ -5,31 +5,16 @@ param(
     [Parameter(Mandatory = $true)][string] $FolderName = $(throw "Name of folder is required")
 )
 
-try{
-    Write-Host "Creating '$FolderName' directory in file share.."
+$storageAccount = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+$fileShare = Get-AzStorageFile -Context $storageAccount.Context -ShareName $FileShareName
+$fileShareFolders =
+    Get-AzStorageFile -ShareName $FileShareName -Context $storageAccount.Context | 
+        where { $_.GetType().Name -eq "AzureStorageFileDirectory" }
 
-    ## Get the storage account context  
-    $ctx = (Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName).Context  
-
-    ## Create directory  
-    Get-AzStorageShare -Context $ctx -Name $FileShareName | New-AzStorageDirectory -Path $FolderName -ErrorAction Stop
-
-    Write-Host "Directory '$FolderName' has been created.."  
-}
-catch [Microsoft.Azure.Storage.StorageException]
-{
-    if($Error[0].Exception.Message -like "*already exists*")
-    {
-        Write-Host "The specified folder already exists."
-    }
-    else
-    {
-        throw
-    }
-}
-catch
-{
-    $ErrorMessage = $_.Exception.Message
-    Write-Error "Failed to create the directory '$FolderName' in file-share '$FileShareName'. Reason: $ErrorMessage"
-    return $null
-}
+if ($FolderName -in $fileShareFolders.Name) {
+    Write-Host "Azure FileShare storage folder '$FolderName' already exists, skipping"
+} else {
+    Write-Verbose "Creating Azure FileShare storage folder '$FolderName' in file share '$FileShareName'.."
+    New-AzStorageDirectory -Context $storageAccount.Context -ShareName $FileShareName -Path $FolderName
+    Write-Host "Created Azure FileShare storage folder '$FolderName' in file share '$FileShareName'"
+}
```

Comment:
Removed the try/catch-match block to determine whether or not the folder already exists and instead getting the list of existing directories and find one that matches the provided name.

Reply:
Nice idea!